### PR TITLE
chore: Husky, eslint, imports, prettier

### LINKED ISSUE #11: Chore: Fix Husky
```suggestion
          }>
```

_Originally posted by @gracescharf in https://github.com/iprado78/fullstack-iex-stock-ticker/pull/8#discussion_r605736211_

### PR DESCRIPTION
**Pull request by iprado78**
This PR does a few things:

1. Rolls back to an earlier version of Husky
2. Adds eslint (pretty much identical configuration to ReactiveTrader)
3. Fixes eslint warnings
4. Adds an alias for top-level imports @/components @/styles, etc.  You can use that rather than relative imports.
5. Orders imports the way that makes sense to me (we can discuss)
6. Changes the prettier format to 80 char lines (your laptop coding will thank you)

### CODE REVIEW

**Review by gracescharf — DISMISSED** [review]
Amazing! I love the alias for imports

**Comment by dwalker11** [comment]
@iprado78 shouldn't we add a typescript compilation step to our build? During a "pre-push" stage, perhaps? Also, while I love the alias for imports, having a catch-all alias bugs me a bit, but not enough to warrant a change.

File: components/SummaryStats/SummaryStats.tsx [hunk 1]
```diff
@@ -1,12 +1,11 @@
...
-import { bind } from "@react-rxjs/core"
-import { map, switchMap, tap } from "rxjs/operators"
+import { map, switchMap } from "rxjs/operators"
 import { ajax } from "rxjs/ajax"
-import { ISummaryStats } from "../../models/SummaryStats"
+import { bind } from "@react-rxjs/core"
+import { ISummaryStats } from "@/models/SummaryStats"
+import { tickerSelections$ } from "../TickerSearch"
 import { summaryStatsConfig } from "./SummaryStats.config"
+import styles from "./SummaryStats.module.scss"
```
Comment (dwalker11) [inline components/SummaryStats/SummaryStats.tsx:9]: @iprado78 could you explain the order of imports?

**Comment by gracescharf** [comment]
>  Also, while I love the alias for imports, having a catch-all alias bugs me a bit

@dwalker11 what's the risk here? not asking to be argumentative, genuinely curious

File: package.json [hunk 2]
```diff
@@ -6,32 +6,56 @@
...
-      "pre-commit": "format"
+      "pre-commit": "lint-staged"
     }
   },
+  "eslintConfig": {
+    "extends": [
+      "eslint:recommended",
+      "react-app",
+      "react-app/jest"
+    ]
```
Comment (gracescharf) [inline package.json:27]:
```suggestion
  "eslintConfig": {
    "plugins": [
      "jsx-a11y"
    ],
    "extends": [
      "eslint:recommended",
      "react-app",
      "react-app/jest",
      "plugin:jsx-a11y/recommended"
    ]
```

**Comment by dwalker11** [comment]
@gracescharf In this case, I would rather see "@components/TicketSearch" rather than "@/components/TicketSearch". Basically, I think it's better to be explicit rather than implicit (although I can't really back that statement up), and I'm used to seeing top-level imports implemented that way

**Comment by iprado78** [comment]
> @iprado78 shouldn't we add a typescript compilation step to our build? During a "pre-push" stage, perhaps? Also, while I love the alias for imports, having a catch-all alias bugs me a bit, but not enough to warrant a change.

@dwalker11  = Re: TypeScript - Next.js build config (which we are really just extending) should already do this.  Are you seeing it compile with TypeScript errors?

Edit - nevermind.  TypeScript errors are getting through.  Let me fix.

I could force us to be explicit.  Will likely mean we get errors when we add folders and forget to extend paths configuration, but maybe that's not a bad thing.

> @gracescharf In this case, I would rather see "@components/TicketSearch" rather than "@/components/TicketSearch".

@gracescharf - Problem is that is ambiguous with external Node scoped packages (e.g., @react-rxjs).  The slash tells you it's part of the app.

**Review by dwalker11 — APPROVED** [review]
Looks good to me

**Pull request merged**